crates/oxc_semantic/src/builder.rs

```diff
@@ -1845,13 +1845,10 @@ impl<'a> SemanticBuilder<'a> {
         /* cfg */
 
         match kind {
-            AstKind::ExportNamedDeclaration(decl) => {
```

Comment:
Should I also remove `AstKind::ExportNamedDeclaration` in leave kind?

Reply:
It seems to work as well if you revert this and remove `|| matches!(self.nodes.parent_kind(self.current_node_id), Some(AstKind::ExportNamedDeclaration(decl)) if decl.export_kind.is_type())` in `ExportSpecifier`

Reply:
Feel free to modify the PR directly if you think it's more efficient.